### PR TITLE
Add YOLO v4 model

### PR DESCRIPTION
**Pull request by eizamaliev**
TODO:

 - [x] Link for model downloading
 - [x] Perfomance metrics
 - [x] Accuracy metrics

### CODE REVIEW

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+models:
```
Comment (vladimir-dudnik) [inline models/public/yolo-v4-tf/accuracy-check.yml:1]: please add link in tools/accuracy-checker/configs

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            ignore_difficult: true
+            presenter: print_scalar
+          - type: coco_precision
+            max_detections: 100
+            threshold: 0.5
+  - name: yolo_v4
+    launchers:
+      - framework: dlsdk
+        model:   FP32/yolo-v4.xml
+        weights: FP32/yolo-v4.bin
```
Comment (eaidova) [inline models/public/yolo-v4-tf/accuracy-check.yml:61]: these 2 lines should be removed

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          threshold: 0.001
+          anchor_masks: [[0,1,2],[3,4,5],[6,7,8]]
+          raw_output: True
+          outputs:
+            - conv2d_93/BiasAdd/Add
+            - conv2d_101/BiasAdd/Add
+            - conv2d_109/BiasAdd/Add
+    datasets:
+      - name: ms_coco_detection_80_class_without_background
+        subsample_size: 100
```
Comment (eaidova) [inline models/public/yolo-v4-tf/accuracy-check.yml:77]: please remove subsample_size

File: tools/accuracy_checker/accuracy_checker/launcher/tf_launcher.py [hunk 4]
```diff
@@ -112,6 +112,7 @@ def predict(self, inputs, metadata=None, **kwargs):
             if metadata is not None:
                 for meta_ in metadata:
                     meta_['input_shape'] = self.inputs_info_for_meta()
+                    meta_['input_shape'].update({name: data.shape for name, data in infer_input.items()})
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/launcher/tf_launcher.py:115]:
```suggestion
                    meta_['input_shape'].update({name: data.shape for name, data in infer_input.items()})
```

File: models/public/yolo-v4-tf/yolo-v4-tf.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
+# yolo-v4-tf
+
+## Use Case and High-Level Description
+
+YOLO v4 is a real-time object detection model implemented with Keras\* from this [repository](https://github.com/david8862/keras-YOLOv3-model-set) and converted to TensorFlow\* framework. This model was pretrained on COCO\* dataset with 80 classes.
+
+## Conversion
+
+1. Download or clone the official [repository](https://github.com/david8862/keras-YOLOv3-model-set) (tested on `ffede5` commit).
```
Comment (IRDonch) [inline models/public/yolo-v4-tf/yolo-v4-tf.md:9]: This commit seems suspect. The README file in this version doesn't even mention YOLOv4.

**Review by IRDonch — APPROVED** [review]

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+          - type: clip_boxes
+            apply_to: prediction
+        metrics:
+          - type: map
+            integral: 11point
+            ignore_difficult: true
+            presenter: print_scalar
+          - type: coco_precision
+            max_detections: 100
+            threshold: 0.5
```
Comment (eaidova) [inline models/public/yolo-v4-tf/accuracy-check.yml:55]: why here only 2 metrics, while in the next section 3?

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  - name: yolo_v4
```
Comment (eaidova) [inline models/public/yolo-v4-tf/accuracy-check.yml:56]:
```suggestion

  - name: yolo_v4
```

File: models/public/yolo-v4-tf/accuracy-check.yml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+          - type: map
+            integral: 11point
+            ignore_difficult: true
+            presenter: print_scalar
+          - type: coco_precision
+            max_detections: 100
+            threshold: 0.5
+          - type: coco_precision
+            max_detections: 100
+            threshold: '0.5:0.05:0.95'
```
Comment (eaidova) [inline models/public/yolo-v4-tf/accuracy-check.yml:97]: It is hard to match metric results after evaluation (you have 2 metrics with the same name, but different values)
```suggestion
          - name: AP@0.5 
            type: coco_precision
            max_detections: 100
            threshold: 0.5
          - name: AP@0.5:0.05:0.95 
            type: coco_precision
            max_detections: 100
            threshold: '0.5:0.05:0.95'
```

**Review by vladimir-dudnik — COMMENTED** [review]
tested all yolo related demos, no one adopted to yolo-v4 yet

File: models/public/yolo-v4-tf/accuracy-check.yml (hunk 6 above)
Comment (eaidova) [reply models/public/yolo-v4-tf/accuracy-check.yml:56]: still no explanation where AP0.5:0.05.:0.95 for tf launching?

File: models/public/yolo-v4-tf/yolo-v4-tf.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
+# yolo-v4-tf
+
+## Use Case and High-Level Description
+
+YOLO v4 is a real-time object detection model based on ["YOLOv4: Optimal Speed and Accuracy of Object Detection"](https://arxiv.org/abs/2004.10934) paper It was implemented with Keras\* from this [repository](https://github.com/david8862/keras-YOLOv3-model-set) and converted to TensorFlow\* framework. This model was pretrained on COCO\* dataset with 80 classes.
```
Comment (eaidova) [inline models/public/yolo-v4-tf/yolo-v4-tf.md:5]:
```suggestion
YOLO v4 is a real-time object detection model based on ["YOLOv4: Optimal Speed and Accuracy of Object Detection"](https://arxiv.org/abs/2004.10934) paper. It was implemented with Keras\* from this [repository](https://github.com/david8862/keras-YOLOv3-model-set) and converted to TensorFlow\* framework. This model was pretrained on COCO\* dataset with 80 classes.
```

Comment (eaidova) [reply models/public/yolo-v4-tf/yolo-v4-tf.md:5]: not sure that part about keras grammatically correct, sounds like it is link on fork of keras, not repo with model training

**Comment by IRDonch** [comment]
Now that #1657 is merged, will you redo `model.yml` to use the original model files?

**Review by IRDonch — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Comment by IRDonch** [comment]
Jenkins please retry a build

**Comment by IRDonch** [comment]
@eizamaliev Please make the pre-convert script executable.

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**